jctools-experimental/src/main/java/org/jctools/sets/SingleWriterHashSet.java

```diff
@@ -268,9 +268,8 @@ public E next() {
         private void findNextVal() {
             E[] array = buffer;
             int i = nextValIndex;
-            E e = null;
             for (; i < array.length; i++) {
-                e = array[i];
+                E e = lvElement(array, calcElementOffset(i, array.length - 1));
```

Comment:
I do not think  required. The buffer is read with a volatile load when the iterator is created. Future reads are ordered with that 'snapshot' read. The semantics here are not very clear for concurrent sets, technically giving you a copy to iterate on for reading is perfectly valid, and so is letting you see further changes to the set which happen concurrently.